Matchmaker role

imadtg opened the pull request:
hopefully this is the last PR, have done some testing and it works just fine

belguawhale committed:
Merge pull request #18 from belguawhale/dev

Oracle, augur roles and a ton of game improvements

imadtg committed:
Matchmaker role

hopefully this is the last PR fr this role lmao

belguawhale commented on bot.py:
```diff
@@ -385,6 +385,8 @@ def check(m):
         if len(session[1]) == 0:
             await client.change_presence(game=client.get_server(WEREWOLF_SERVER).me.game, status=discord.Status.online)
     await send_lobby(leave_msg)
+    if member in list(session[1]) and session[0]:
```
What's the purpose of this? Member isn't even defined outside of the for loops and each member would have already been removed from the game on line 382

belguawhale commented on bot.py:
```diff
@@ -847,6 +851,51 @@ def check(m):
+            parameters = parameters.lower().split(" and ")
+            if len(parameters) == 2:
+                player1 = get_player(parameters[0])
+                player2 = get_player(parameters[1])
```
Add an error message if two players were not chosen? (currently it just returns if the command `match 1` is used)

imadtg replied:
done

belguawhale commented on bot.py:
```diff
@@ -2132,7 +2181,8 @@ def win_condition():
     
     for player in session[1]:
-        if get_role(player, 'actualteam') == win_team:
+        lover = o.split(':')[1]
```
What's o? Doesn't seem to be defined above

imadtg replied:
oh wait

imadtg replied:
fixed

belguawhale commented on bot.py:
```diff
@@ -2935,6 +3015,14 @@ def sort_roles(role_list):
+                    #fix for lover suicide message appears if player dies even tho lover died already
+                    session[1][player][4].remove('lover:' + lover)
+                    session[1][lover][4].remove('lover:' + player)
+                    await player_death(player, 'night kill')
```
uhh doesn't this make night kills not cause the other lover to die?

imadtg replied:
this only happens if both lovers were killed at night, and that is a fix for this 
![tempae](https://user-images.githubusercontent.com/30117228/29088134-1fb73898-7c70-11e7-8d11-9f361dbac112.PNG)

belguawhale commented on bot.py:
```diff
@@ -3132,7 +3220,8 @@ def sort_roles(role_list):
                      'observe' : ['werecrow', 'sorcerer'],
                      'pass' : ['harlot', 'hunter'],
-                     'id' : ['detective']}
+                     'id' : ['detective']
```
I think you're missing a comma here

belguawhale commented on bot.py:
```diff
@@ -3212,7 +3305,9 @@ def sort_roles(role_list):
-            [0, 0, 0, 0, 0, 0,  1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}
+            [0, 0, 0, 0, 0, 0,  1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+            'matchmaker' :
+            [0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}
```
Did you `!verifygamemodes` to make sure role counts are correct?

imadtg replied:
fixed too

belguawhale commented on bot.py:
```diff
@@ -667,6 +669,8 @@ def check(m):
                                    "`{}role gunner` for more information.".format(
                             session[1][player][4].count('bullet'), '' if session[1][player][4].count('bullet') == 1 else 's',
                             BOT_PREFIX))
+                    if role == 'matchmaker' and sendrole:
```
You can add matchmaker to the list of roles on line 664 and this will automatically be factored in

imadtg replied:
but i want it to only be dmed once

belguawhale replied:
Oh, right

belguawhale replied:
Personally i'd still like to be dmed the list of living players even after my ability is used, like hunter but that's my personal opinion

imadtg committed:
minor bugfixes

Sorry!

imadtg committed:
last bits

imadtg committed:
bug fix

fixed a bug causing lover suicide message to be shown on daybreak even if lover died in night

imadtg committed:
bug fixes

fixed bugs with !fleave all (lovers committing suicide, win condition breaking)

imadtg committed:
More bug fixes

fixed mm not getting match other flag on game start

imadtg committed:
fool lovers

now lovers of the lynched fool win with him too, not implemented into gamemodes.

imadtg committed:
tweaks

added error message, and first night checking

imadtg committed:
added 'match' an alias to 'choose'

belguawhale commented on bot.py:
```diff
@@ -3027,7 +3141,15 @@ def sort_roles(role_list):
+                            lovers = o
+                        else:
+                            lovers = []
+                        await end_game(win_msg, [lynched_player] + lovers)
```
If matched fool is lynched, their lover doesn't win except in random gamemode. Just for the future

imadtg replied:
it isn't like fool and matchmaker have mutual gamemodes other then random for now \:P

imadtg committed:
decent tweaks to matchmaker and lover mechanics

lovers now can steal win if there are no non lover alive
fool now wins with his first level lover (only possible in random mode anyways)
matchmakers forgetting to match doesn't accidentally match two couples that already are matched, however, a matchmaker can intentionally match an already existing couple

imadtg committed:
Merge pull request #1 from belguawhale/dev

Added !wait command and some fixes

belguawhale committed:
Matchmaker update!

Fixed some bugs and streamlined cmd_choose and end of game stats

Fixed role order with matchmaker

Replaced shaman in orgy gamemode with matchmaker

Added valentines gamemode

Fixed a bug where matchmaker would be automatically forced to end night

imadtg committed:
Merge pull request #2 from belguawhale/imadtg-master

Matchmaker update!

belguawhale commented:
Matchmaker role! (thanks to imadtg)
